dirsrvtests/tests/suites/logging/access_json_logging_test.py

```diff
@@ -560,6 +564,88 @@ def test_access_json_format(topo_m2, setup_test):
     assert event['oid_name'] == "REPL_END_NSDS50_REPLICATION_REQUEST_OID"
     assert event['name'] == "replication-multisupplier-extop"
 
+    #
+    # TLS INFO/TLS CLIENT INFO
+    #
+    RDN_TEST_USER = 'testuser'
+    RDN_TEST_USER_WRONG = 'testuser_wrong'
+    inst.enable_tls()
+    inst.restart()
+
+    users = UserAccounts(inst, DEFAULT_SUFFIX)
+    user = users.create(properties={
+        'uid': RDN_TEST_USER,
+        'cn': RDN_TEST_USER,
+        'sn': RDN_TEST_USER,
+        'uidNumber': '1000',
+        'gidNumber': '2000',
+        'homeDirectory': f'/home/{RDN_TEST_USER}'
+    })
+
+    ssca_dir = inst.get_ssca_dir()
+    ssca = NssSsl(dbpath=ssca_dir)
+    ssca.create_rsa_user(RDN_TEST_USER)
+    ssca.create_rsa_user(RDN_TEST_USER_WRONG)
+
+    # Get the details of where the key and crt are.
+    tls_locs = ssca.get_rsa_user(RDN_TEST_USER)
+    tls_locs_wrong = ssca.get_rsa_user(RDN_TEST_USER_WRONG)
+
+    user.enroll_certificate(tls_locs['crt_der_path'])
+
+    # Turn on the certmap.
+    cm = CertmapLegacy(inst)
+    certmaps = cm.list()
+    certmaps['default']['DNComps'] = ''
+    certmaps['default']['FilterComps'] = ['cn']
+    certmaps['default']['VerifyCert'] = 'off'
+    cm.set(certmaps)
+
+    # Check that EXTERNAL is listed in supported mechns.
+    assert (inst.rootdse.supports_sasl_external())
+
+    # Restart to allow certmaps to be re-read: Note, we CAN NOT use post_open
+    # here, it breaks on auth. see lib389/__init__.py
+    inst.restart(post_open=False)
+
+    # Attempt a bind with TLS external
+    inst.open(saslmethod='EXTERNAL', connOnly=True, certdir=ssca_dir,
+              userkey=tls_locs['key'], usercert=tls_locs['crt'])
+    inst.restart()
+
+    event = get_log_event(inst, "TLS_INFO")
+    assert event is not None
+    assert 'tls_version' in event
+    assert 'keysize' in event
+    assert 'cipher' in event
+
+    event = get_log_event(inst, "TLS_CLIENT_INFO",
+                          "subject",
+                          "CN=testuser,O=testing,L=389ds,ST=Queensland,C=AU")
+    assert event is not None
+    assert 'tls_version' in event
+    assert 'keysize' in event
+    assert 'issuer' in event
+
+    event = get_log_event(inst, "TLS_CLIENT_INFO",
+                          "client_dn",
+                          "uid=testuser,ou=People,dc=example,dc=com")
+    assert event is not None
+    assert 'tls_version' in event
+    assert event['msg'] == "client bound"
+
+    # Check for failed certmap error
+    with pytest.raises(ldap.INVALID_CREDENTIALS):
+        inst.open(saslmethod='EXTERNAL', connOnly=True, certdir=ssca_dir,
+                  userkey=tls_locs_wrong['key'],
+                  usercert=tls_locs_wrong['crt'])
+
+    event = get_log_event(inst, "TLS_CLIENT_INFO", "err", -185)
```

Comment:
I think it would be better to describe the magic number `-185`, as it's unclear in the test what exactly the error is.